### PR TITLE
add basic progress history for user

### PR DESCRIPTION
**Pull request by joelhooks**
This adds **rudimentary** course history in what I'd describe as a brute force sort of way to `/user`

I limited it to 5, but the backing data is paginated. My graphql query is the "brute force" aspect of it and we could really stop and consider what progress looks like in gql. cc @theianjones on that one big time

image

### CODE REVIEW

File: src/components/pages/home/in-progress-collection.tsx [hunk 1]
```diff
@@ -15,20 +15,25 @@ const InProgressCollection: FunctionComponent<any> = ({collection}) => {
     series,
     slug,
     collection_progress,
+    progress,
     type,
     path,
     items = [],
   } = collection
 
+  const definedProgress = collection_progress || progress
```
Comment (joelhooks) [inline src/components/pages/home/in-progress-collection.tsx:24]: disparity between REST and gql apis

File: src/lib/users.ts [hunk 2]
```diff
@@ -10,3 +16,107 @@ export async function loadCurrentUser(
...
   return user
 }
+
+export async function loadUserProgress(
+  user_id: number,
+  page = 1,
+  per_page = 5,
+  token?: string,
+): Promise<any> {
+  const query = gql`
```
Comment (joelhooks) [inline src/lib/users.ts:26]: this query feels like it's way more beefy than it needs to be but I was just trying to get it done and on the page so we can start to consider it

File: src/pages/user/index.tsx [hunk 3]
```diff
@@ -30,11 +32,23 @@ const User: React.FunctionComponent<
...
   const {stripe_customer_id} = account
   const {viewer, authToken} = useViewer()
+  const [progress, setProgress] = React.useState<any>([])
 
   const {email: currentEmail, accounts, providers} = viewer || {}
   const {slug} = get(accounts, '[0]', {})
   const isConnectedToGithub = providers?.includes('github')
 
+  React.useEffect(() => {
+    const loadProgressForUser = async (user_id: number) => {
```
Comment (joelhooks) [inline src/pages/user/index.tsx:42]: I didn't paginate this, but the data is paginated and limited to 5 right now. We could make it "infinite scroll" or something like that if we wanted.

File: src/pages/user/index.tsx [hunk 4]
```diff
@@ -83,6 +98,16 @@ const User: React.FunctionComponent<
             stripeCustomerId={stripe_customer_id}
             slug={slug}
           />
+          <div>
+            {progress.map((item) => {
```
Comment (joelhooks) [inline src/pages/user/index.tsx:102]: I don't know if this page is the perfect spot for this, but made sense for now

File: src/components/pages/home/in-progress-collection.tsx (hunk 1 above)
Comment (jbranchaud) [reply src/components/pages/home/in-progress-collection.tsx:24]: Could these be normalized in `src/lib/users.ts` and then this component doesn't need to know about API layer concerns?

E.g. before returning `all_progress` from `loadUserProgress`, you could transform it, something like this:

```javascript
// all the graphql fetching stuff ...

const normalizedData = all_progress.data.map(item => {
  const { progress, ...rest } = item;
  return { ...rest, collection_progress: progress }
})
const normalizedAllProgress = { ...all_progress, data: normalizedData };

return normalizedAllProgress;
```

Comment (joelhooks) [reply src/components/pages/home/in-progress-collection.tsx:24]: they could be normalized in several places above this as another project

**Review by jbranchaud — APPROVED** [review]
It doesn't seem all that rudimentary to me, looks good 👍

**Review by theianjones — APPROVED** [review]
Looks like theres some conflicts with `yarn.lock` file

**Pull request merged**